functests/ipv6/ipv6.go

```diff
@@ -47,51 +48,68 @@ var _ = Describe("sriov", func() {
-			_, err := client.Client.Pods(testNamespace).Create(podDefinition)
-			Eventually(func() k8sv1.PodPhase {
-				runningPod, err := client.Client.Pods(testNamespace).Get(pingedPodName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return runningPod.Status.Phase
-			}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
-			pod, err := client.Client.Pods(testNamespace).Get(pingedPodName, metav1.GetOptions{})
-			var nets []Network
-			err = json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/networks-status"]), &nets)
-			Expect(err).ToNot(HaveOccurred())
-			for _, net := range nets {
-				if net.Interface != sriovInterfaceName {
-					continue
-				}
-				for i, ip := range net.Ips {
-					pingPod(ip, i, nodeSelector)
-				}
-				return
+	DescribeTable("sriov", func(podNamePrefix string, netAttachment string, expectedNicCount int) {
+		var _ = Context("IPv6 configured secondary interfaces on pods", func() {
+			pod := createTestPod(podNamePrefix+pingedPodName, []string{"/bin/bash", "-c", "--"},
+				[]string{"while true; do sleep 300000; done;"}, nodeSelector, netAttachment)
+			ips := getSriovNicIps(pod)
+			Expect(ips).NotTo(BeNil(), "No sriov network interface found.")
+			Expect(len(ips)).Should(Equal(expectedNicCount))
+			for _, ip := range ips {
+				pingPod(podNamePrefix+pingingPodName, ip, nodeSelector, netAttachment)
 			}
-			Fail("No sriov network inerface found.")
 		})
-	})
+	},
+		Entry("IPv6 configured secondary interfaces on pods", "dual-stack-", dualStackSriovNetworkAttachment, 2),
+		Entry("Should be able to ping each other in a IPv6 only configuration", "ipv6only-", ipv6onlySriovNetworkAttachment, 1),
+	)
 })
 
-func pingPod(ip string, podNumber int, nodeSelector string) {
-	podName := fmt.Sprintf("%s-%d", pingingPodName, podNumber)
-	podDefinition := dualStackTestPod(podName, []string{"sh", "-c", "ping -c 3 " + ip}, []string{}, nodeSelector)
-	_, err := client.Client.Pods(testNamespace).Create(podDefinition)
+func getSriovNicIps(pod *k8sv1.Pod) []string {
+	var nets []Network
+	err := json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/networks-status"]), &nets)
+	Expect(err).ToNot(HaveOccurred())
+	for _, net := range nets {
+		if net.Interface != sriovInterfaceName {
+			continue
+		}
+		return net.Ips
+	}
+	return nil
+}
+
+func createTestPod(name string, command []string, args []string, node string, sriovNetworkAttachment string) *k8sv1.Pod {
+	podDefinition := testPodDefintion(name, []string{"/bin/bash", "-c", "--"},
+		[]string{"while true; do sleep 300000; done;"}, node, sriovNetworkAttachment)
+	createdPod, err := client.Client.Pods(testNamespace).Create(podDefinition)
+
+	Eventually(func() k8sv1.PodPhase {
+		runningPod, err := client.Client.Pods(testNamespace).Get(createdPod.ObjectMeta.Name, metav1.GetOptions{})
```

Comment:
You can do createdPod.Name as the ObjectMeta is embedded

Reply:
done